### PR TITLE
fix pagination for absolute body overflow

### PR DESCRIPTION
**Pull request by mitsuru**
## 概要
- body直下の `position:absolute` subtree がページをまたぐ場合に、交差する各ページへpagination geometryを生成するようにしました。
- absolute overflowで増えたページにも `position:fixed` が繰り返し描画されるよう、absolute pass後にfixed fragmentsを再生成します。
- `css/css-page/monolithic-overflow-013-print.html` をPASSへ昇格しました。

## 設計
- 既存どおり `fragment_pagination_root` はOOF childをskipし、body直下absolute用の補助passでgeometryを補完します。
- absolute subtree内のin-flow descendantsへ、親fragmentが増やしたpage budgetを伝播します。
- `contain:size` のmonolithic child後続テキストに対し、親背景のcontinuation stripとtext fragmentが同じpage-local位置に出るようにしています。

## 検証
- `cargo fmt --check`
- `cargo test -p fulgur position_absolute_body_direct --lib`
- `FONTCONFIG_FILE="$PWD/examples/.fontconfig/fonts.conf" cargo run -p fulgur-wpt --example run_one --release -- target/wpt/css/css-page/monolithic-overflow-013-print.html`

## WPT delta
| Test | Before | After |
| --- | --- | --- |
| `css/css-page/monolithic-overflow-013-print.html` | FAIL | PASS |

Refs: fulgur-z4zc

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected page-count and overflow handling so absolutely positioned content spans and repeats across all intersecting pages.
  * Ensured fixed-element repetition remains correct when body-direct absolute content increases the effective page count.
  * Resolved a failing CSS print test; overall pass count increased by one.

* **Tests**
  * Added coverage for multi-page absolute geometry, overflow placement, fixed-element repetition, and pagination edge cases.

### CODE REVIEW

File: crates/fulgur/src/pagination_layout.rs [hunk 1]
```diff
@@ -2647,8 +2661,32 @@ fn record_subtree_fragments_at_offset(
...
+            .filter_map(|child_id| doc.get_node(*child_id))
+            .filter(|child| {
+                child.primary_styles().is_some_and(|s| {
+                    s.get_box()
+                        .clone_contain()
+                        .contains(::style::values::computed::box_::Contain::SIZE)
+                })
+            })
+            .map(|child| (child.final_layout.size.height - page_h_px).max(0.0))
+            .sum();
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/pagination_layout.rs:2700]: cbbfaf3 で修正しました。`monolithic_adjust` の prepass で `position:absolute` / `position:fixed` の子を除外し、実際の subtree walk が skip する OOF child と同じ扱いにしました。あわせて body に in-flow content がある場合は body-direct absolute overflow だけで page count を増やさないようにし、CI 失敗の `abs_positioned_does_not_force_extra_pages_for_short_flow` も検証済みです。

File: crates/fulgur-wpt/expectations/css-page.txt [hunk 2]
```diff
@@ -100,7 +100,7 @@ FAIL  css/css-page/monolithic-overflow-009-print.html  # page 1 diff: 673864/891
 FAIL  css/css-page/monolithic-overflow-010-print.html  # page 1 diff: 673864/891662 pixels exceed tol (max_ch=255)
 FAIL  css/css-page/monolithic-overflow-011-print.html  # page 1 diff: 673864/891662 pixels exceed tol (max_ch=255)
 PASS  css/css-page/monolithic-overflow-012-print.html
-FAIL  css/css-page/monolithic-overflow-013-print.html  # page count mismatch: test=1 ref=2
+PASS  css/css-page/monolithic-overflow-013-print.html
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur-wpt/expectations/css-page.txt:103]: cbbfaf3 で修正しました。`monolithic-overflow-013-print.html` の昇格に合わせて `css-page.txt` の summary を `79 PASS, 144 FAIL, 34 SKIP` に更新しました。

File: crates/fulgur/src/pagination_layout.rs [hunk 3]
```diff
@@ -2561,6 +2560,17 @@ pub fn append_position_absolute_body_direct_fragments(
...
+        let Some(child) = doc.get_node(child_id) else {
+            return false;
+        };
+        if let Some(text) = child.text_data()
+            && text.content.chars().all(char::is_whitespace)
+        {
+            return false;
+        }
+        !is_out_of_flow_positioned(child)
+    });
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/pagination_layout.rs:2577]: Fixed in commit a3788a9. Added running_store: Option<&RunningElementStore> parameter to append_position_absolute_body_direct_fragments and pass Some(&running_store) from the production call site in engine.rs. The body_has_in_flow_content closure now skips running nodes. Added a unit test covering this path.

File: crates/fulgur/src/pagination_layout.rs [hunk 4]
```diff
@@ -2658,45 +2711,67 @@ fn record_subtree_fragments_at_offset(
...
+                && last_page_f.is_finite()
+                && first_page_f <= last_page_f
+                && first_page_f < total_pages as f32
+            {
+                let first_page = first_page_f as u32;
+                let last_page = if may_extend_pages {
+                    last_page_f as u32
+                } else {
+                    (last_page_f as u32).min(total_pages.saturating_sub(1))
+                };
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/pagination_layout.rs:2749]: Fixed in commit a3788a9. Changed the guard to (may_extend_pages || first_page_f < total_pages as f32) so absolute subtrees starting beyond the current page budget still emit fragments when may_extend_pages is set. Added a unit test: empty body with position:absolute top:1600px height:100px, total_pages=1, viewport_h=800 must emit a fragment on page 2.
